{
	"id": "CVE-2024-56764",
	"sourceIdentifier": "416baaa9-dc9f-4396-8d5f-8c081fb06d67",
	"vulnStatus": "Awaiting Analysis",
	"published": "2025-01-06T17:15:42.483",
	"lastModified": "2025-01-06T17:15:42.483",
	"descriptions": [
		{
			"lang": "en",
			"value": "In the Linux kernel, the following vulnerability has been resolved:\n\nublk: detach gendisk from ublk device if add_disk() fails\n\nInside ublk_abort_requests(), gendisk is grabbed for aborting all\ninflight requests. And ublk_abort_requests() is called when exiting\nthe uring context or handling timeout.\n\nIf add_disk() fails, the gendisk may have been freed when calling\nublk_abort_requests(), so use-after-free can be caused when getting\ndisk's reference in ublk_abort_requests().\n\nFixes the bug by detaching gendisk from ublk device if add_disk() fails."
		},
		{
			"lang": "es",
			"value": "En el kernel de Linux, se ha resuelto la siguiente vulnerabilidad: ublk: separar gendisk del dispositivo ublk si add_disk() falla Dentro de ublk_abort_requests(), gendisk se toma para abortar todas las solicitudes en vuelo. Y ublk_abort_requests() se llama al salir del contexto uring o al gestionar el tiempo de espera. Si add_disk() falla, es posible que gendisk se haya liberado al llamar a ublk_abort_requests(), por lo que se puede producir un use-after-free al obtener la referencia del disco en ublk_abort_requests(). Corrige el error separando gendisk del dispositivo ublk si add_disk() falla."
		}
	],
	"references": [
		{
			"source": "416baaa9-dc9f-4396-8d5f-8c081fb06d67",
			"url": "https://git.kernel.org/stable/c/75cd4005da5492129917a4a4ee45e81660556104"
		},
		{
			"source": "416baaa9-dc9f-4396-8d5f-8c081fb06d67",
			"url": "https://git.kernel.org/stable/c/7d680f2f76a3417fdfc3946da7471e81464f7b41"
		}
	],
	"metrics": {}
}
